port/unix/omrtime.c

```diff
@@ -161,7 +161,14 @@ omrtime_hires_clock(struct OMRPortLibrary *portLibrary)
 {
 #if defined(OSX)
 	return clock_gettime_nsec_np(CLOCK_MONOTONIC_RAW);
-#else /* defined(OSX) */
+#elif defined(LINUX) /* defined(OSX) */
+	uint64_t ret = 0;
+	struct timespec ts;
+	if (0 == clock_gettime(CLOCK_MONOTONIC_RAW, &ts)) {
+		ret = ((uint64_t)ts.tv_sec * OMRTIME_NANOSECONDS_PER_SECOND) + (uint64_t)ts.tv_nsec;
```

Comment:
There are signed to unsigned conversions: potential risk of data loss. There are existing instances where this is done. @keithc-ca @tajila Do we accept the risk or shall I open an issue to address this in the future?

Reply:
`omrtime_current_time_nanos()` in the same file does the same conversion.
https://github.com/eclipse/omr/blob/7ebc813b2fe834208515bb8c44de0cf8d6404db4/port/unix/omrtime.c#L107

Reply:
> There are also existing instances where this is done.

Yes, I was referring to the above change. The same issue also exists in `omrtime_current_time_nanos()`. It is most likely that this PR will be merged as-is, and a new issue will be opened for the signed to unsigned conversion. Just waiting for confirmation from either @tajila or @keithc-ca.

Reply:
`int64_t` is the correct choice to store the values returned by the underlying OS functions. I didn't find any documentation, which justifies the usage of `uint64_t`. I speculate that `uint64_t` was used because time can't be negative and the underlying behaviour of the OS API was ignored. I have opened https://github.com/eclipse/omr/issues/7042 to address this issue.